cmd/csi/app/csi.go

```diff
@@ -0,0 +1,203 @@
+	if err := startCmd.MarkFlagRequired("nodeid"); err != nil {
+		ErrorAndExit(err)
+	}
+
+	startCmd.Flags().StringVarP(&endpoint, "endpoint", "", "", "CSI endpoint")
+	if err := startCmd.MarkFlagRequired("endpoint"); err != nil {
+		ErrorAndExit(err)
+	}
+
+	startCmd.Flags().StringVarP(&metricsAddr, "metrics-addr", "", ":8080", "The address the metrics endpoint binds to.")
+	startCmd.Flags().StringVarP(&pprofAddr, "pprof-addr", "", "", "The address for pprof to use while exporting profiling results")
+	startCmd.Flags().AddGoFlagSet(flag.CommandLine)
+
+	// start csi manager
+	startCmd.Flags().BoolVar(&enableManager, "enable-manager", false, "Enable manager to recover failed mount point automatically")
+	startCmd.Flags().IntVar(&managerPeriod, "manager-period", 10, "CSI manager sync pods period, in seconds")
+	startCmd.Flags().StringVar(&clientCert, "client-cert", "", "Kubelet client cert")
+	startCmd.Flags().StringVar(&clientKey, "client-key", "", "Kubelet client key")
+	startCmd.Flags().IntVar(&timeout, "timeout", 10, "Kubelet client timeout")
+	nodeIp = os.Getenv("NODE_IP")
+}
+
+func ErrorAndExit(err error) {
+	fmt.Fprintf(os.Stderr, "%s", err.Error())
+	os.Exit(1)
+}
+
+func handle() {
+	// startReaper()
+	fluid.LogVersion()
+
+	if pprofAddr != "" {
+		glog.Infof("Enabling pprof with address %s", pprofAddr)
+		mux := http.NewServeMux()
+		mux.HandleFunc("/debug/pprof/", pprof.Index)
+		mux.HandleFunc("/debug/pprof/cmdline", pprof.Cmdline)
+		mux.HandleFunc("/debug/pprof/profile", pprof.Profile)
+		mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
+		mux.HandleFunc("/debug/pprof/trace", pprof.Trace)
+
+		pprofServer := http.Server{
+			Addr:    pprofAddr,
+			Handler: mux,
+		}
+		glog.Infof("Starting pprof HTTP server at %s", pprofServer.Addr)
+
+		go func() {
+			go func() {
+				ctx := context.Background()
+				<-ctx.Done()
+
+				ctx, cancelFunc := context.WithTimeout(context.Background(), 60*time.Minute)
+				defer cancelFunc()
+
+				if err := pprofServer.Shutdown(ctx); err != nil {
+					glog.Error(err, "Failed to shutdown debug HTTP server")
+				}
+			}()
+
+			if err := pprofServer.ListenAndServe(); !errors.Is(http.ErrServerClosed, err) {
+				glog.Error(err, "Failed to start debug HTTP server")
+				panic(err)
+			}
+		}()
+	}
+
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:             scheme,
+		MetricsBindAddress: metricsAddr,
+		Port:               9443,
+	})
+
+	if err != nil {
+		panic(fmt.Sprintf("csi: unable to create controller manager due to error %v", err))
+	}
+
+	ctx := ctrl.SetupSignalHandler()
+	go func() {
+		if err := mgr.Start(ctx); err != nil {
+			panic(fmt.Sprintf("unable to start controller manager due to error %v", err))
+		}
+	}()
+
+	if enableManager {
+		manageStart(mgr.GetClient())
+	}
+
+	d := csi.NewDriver(nodeID, endpoint, mgr.GetClient())
+	d.Run()
+}
+
+func manageStart(k8sClient client.Client) {
+	glog.V(3).Infoln("start csi manager")
```

Comment:
I suggest this function should return error if it's not able to start recovery manager.